Jupyter Notebooks/LUVOIR/fields_building_numerical_all_laurent.ipynb

```diff
@@ -0,0 +1,928 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "divine-coordinator",
+   "metadata": {},
+   "source": [
+    "# This notebook is for testing and Integration of fields_building_numerical_all.py with the develop branch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "induced-charles",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import time\n",
+    "import functools\n",
+    "from shutil import copy\n",
+    "import matplotlib.pyplot as plt\n",
+    "from astropy.io import fits\n",
+    "import astropy.units as u\n",
+    "import hcipy as hc\n",
+    "import numpy as np\n",
+    "import pastis.util as util    \n",
+    "from pastis.config import CONFIG_PASTIS \n",
+    "from pastis.e2e_simulators.luvoir_imaging import LuvoirA_APLC "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "engaged-thriller",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "root_dir = CONFIG_PASTIS.get('local', 'local_data_path')\n",
+    "output_dir = CONFIG_PASTIS.get('local', 'output_data_folder') #needs to be added to config_ini\n",
+    "coronagraph_design = 'small' #c user provides\n",
+    "overall_dir = output_dir + 'luvoirA-'+ coronagraph_design \n",
+    "resDir = os.path.join(overall_dir, 'matrix_numerical')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "searching-regular",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_LO = CONFIG_PASTIS.getint('calibration', 'max_LO') \n",
+    "max_MID = CONFIG_PASTIS.getint('calibration', 'max_MID')\n",
+    "max_HI = CONFIG_PASTIS.getint('calibration', 'max_HI')\n",
+    "num_DM_act = CONFIG_PASTIS.getint('calibration', 'num_DM_act') "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "described-acoustic",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nb_seg = CONFIG_PASTIS.getint('LUVOIR', 'nb_subapertures')\n",
+    "wvln = CONFIG_PASTIS.getfloat('LUVOIR', 'lambda') * 1e-9  # m\n",
+    "diam = CONFIG_PASTIS.getfloat('LUVOIR', 'diameter')  # m\n",
+    "nm_aber = CONFIG_PASTIS.getfloat('calibration', 'single_aberration') * 1e-9   # m"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adjusted-mount",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "im_lamD = CONFIG_PASTIS.getfloat('numerical', 'im_size_lamD_hcipy')  # image size in lambda/D\n",
+    "sampling = CONFIG_PASTIS.getfloat('numerical', 'sampling')"
```

Comment:
If you deleted `sampling` from the section `numerical` in the configfile (which you should have and you did), this will break. You will need to read the sampling from the `LUVOIR` section. Please fix everywhere this shows up.